### PR TITLE
Form context validation

### PR DESCRIPTION
**Pull request by peterjcoffman**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- N/A Updated doc comments / prop-types, or determined not required.
- N/A Reviewed and tested on Mobile, or determined not required.
- N/A Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: desktop/cmp/form/FormField.js [hunk 1]
```diff
@@ -47,8 +47,13 @@ export const [FormField, formField] = hoistCmp.withFactory({
...
         // Resolve FieldModel
-        const formContext = useContext(FormContext),
-            formModel = formContext.model;
+        const formContext = useContext(FormContext);
+        throwIf(
+            !formContext || (Object.entries(formContext).length === 0),
+            'Form field could not find valid FormContext. ' +
+            'Possibly caused by using an HTML form rather than a Hoist form.'
+        );
+        const formModel = formContext.model;
```
Comment (lbwexler) [inline desktop/cmp/form/FormField.js]: + . Doesn't seem like the Form or FormModel is necessary to Render (or should not be difficult to make it so). Think it a gentler development ex

+ Error message should be clearer, reference the actual package e.g.  "Make sure you are using a Hoist form ('@xh/hoist/cmp/form/Form')  and not an Html Form ('@xh/layout/Form`).

Comment (peterjcoffman) [reply desktop/cmp/form/FormField.js]: I addressed those two things.

Comment (amcclain) [reply desktop/cmp/form/FormField.js]: Minor style point. would recommend using backticks here for the multi-line string with inner quotation marks.

Comment (amcclain) [reply desktop/cmp/form/FormField.js]: No need to go back to change this now, but the idea with the backticks is that you don't need to concat with `+` - you can include newlines in backtick quoted strings, so multi-line is cleaner.

**Pull request merged**